### PR TITLE
Add basic-authentication to Kibana via proxy

### PR DESCRIPTION
**Pull request by webwurst**
This adds a new deployment of a caddy webserver. For now it is configured to do basic authentication and if successful to proxy traffic to the kibana service. Username, password are hardcoded as `admin`/`password`. So see this as a proof on concept. You can easily add or remove this proxy or its ingress to enable or deny access to Kibana.

Passwords are not encrypted anyway because of https://github.com/giantswarm/giantswarm/issues/776

### CODE REVIEW

File: manifests/kibana/basic-auth/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: kibana
+        component: basic-auth
+    spec:
+      containers:
+      - name: caddy
+        image: dockermuenster/caddy:0.9.3
```
Comment (marians) [inline manifests/kibana/basic-auth/deployment.yaml]: Relying on https://hub.docker.com/r/dockermuenster/caddy/ seems a bit odd to me, as it meens introducing another party to trust. Maybe copy this to `giantswarm/caddy`?

File: manifests/kibana/basic-auth/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        ports:
+        - containerPort: 80
+          name: http
+          protocol: TCP
+        # - containerPort: 443
+        #   name: https
+        #   protocol: TCP
```
Comment (marians) [inline manifests/kibana/basic-auth/deployment.yaml]: Remove commented YAML?

File: manifests/kibana/basic-auth/ingress.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  name: kibana-basic-auth
+  namespace: logging
+  labels:
+    app: kibana
+    component: basic-auth
+spec:
+  backend:
+    serviceName: kibana-basic-auth
+    servicePort: 80
+    # servicePort: 443
```
Comment (marians) [inline manifests/kibana/basic-auth/ingress.yaml]: Remove commented YAML?

File: manifests/kibana/basic-auth/service.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  type: NodePort
+  ports:
+  - port: 80
+    protocol: TCP
+    targetPort: 80
+    name: http
+  # - port: 443
+  #   protocol: TCP
+  #   targetPort: 443
+  #   name: https
```
Comment (marians) [inline manifests/kibana/basic-auth/service.yaml]: Remove commented YAML?

File: manifests/kibana/deployment.yaml [hunk 5]
```diff
@@ -13,7 +13,7 @@ spec:
       containers:
       - name: kibana
         # image: gcr.io/google_containers/kibana:1.3
```
Comment (marians) [inline manifests/kibana/deployment.yaml]: Remove commented YAML?

**Comment by puja108** [comment]
Agree with @marians above. As talked about in Slack we will move this to it's own repo, so people can reuse the Pod in a variety of use cases and we can adjust functionality without depending on other pods.

**Comment by webwurst** [comment]
Issue moved, so closing this here.

**Pull request closed without merging**